### ISSUE TITLE
assert next_skip_stack is None

### ISSUE THREAD

**Issue by nrshapiro** [Kozea/WeasyPrint#1638]
I've been getting assert next_skip_stack is None failures to convert on some of my files.

I took one that failed, and kept eliminating things until I got a minimum version that would fail.

Then I ran a format and indent in XMLAuthor, and it works.  

So I can't figure out what's causing the problem.

(env) E:\usr3\GitHub\openpubarchive\app>pip show weasyprint
Name: weasyprint
Version: 54.3

```
(env) E:\usr3\GitHub\openpubarchive\app>python -m weasyprint --verbose ANRP.001.0007A.html ANRP.001.0007A.pdf
INFO: Step 1 - Fetching and parsing HTML - ANRP.001.0007A.html
INFO: Step 2 - Fetching and parsing CSS - https://cdn.jsdelivr.net/npm/fork-awesome@1.2.0/css/fork-awesome.min.css
WARNING: Ignored `font-display:block` at 12:478, descriptor not supported.
WARNING: Ignored `text-rendering:auto` at 12:585, unknown property.
WARNING: Ignored `filter:none` at 12:2847, unknown property.
INFO: Step 3 - Applying CSS
INFO: Step 4 - Creating formatting structure
INFO: Step 5 - Creating layout - Page 1
Traceback (most recent call last):
  File "C:\Users\nrsha\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\nrsha\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\__main__.py", line 192, in <module>
    main()
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\__main__.py", line 188, in main
    html.write_pdf(output, **kwargs)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\__init__.py", line 186, in write_pdf
    counter_style=counter_style, image_cache=image_cache)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\__init__.py", line 136, in render
    optimize_size, font_config, counter_style, image_cache)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\document.py", line 930, in _render
    [Page(page_box) for page_box in page_boxes],
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\document.py", line 930, in <listcomp>
    [Page(page_box) for page_box in page_boxes],
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\__init__.py", line 133, in layout_document
    pages = list(make_all_pages(context, root_box, html, pages))
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\page.py", line 855, in make_all_pages
    page, resume_at = remake_page(i, context, root_box, html)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\page.py", line 794, in remake_page
    page_state)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\page.py", line 593, in make_page
    discard=False)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 58, in block_level_layout
    page_is_empty, absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 73, in block_level_layout_switch
    discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 114, in block_box_layout
    absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 646, in block_container_layout
    next_page)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 452, in _in_flow_layout
    fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 58, in block_level_layout
    page_is_empty, absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 73, in block_level_layout_switch
    discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 114, in block_box_layout
    absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 646, in block_container_layout
    next_page)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 452, in _in_flow_layout
    fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 58, in block_level_layout
    page_is_empty, absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 73, in block_level_layout_switch
    discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 114, in block_box_layout
    absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 646, in block_container_layout
    next_page)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 452, in _in_flow_layout
    fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 58, in block_level_layout
    page_is_empty, absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 73, in block_level_layout_switch
    discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 114, in block_box_layout
    absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 646, in block_container_layout
    next_page)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 452, in _in_flow_layout
    fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 58, in block_level_layout
    page_is_empty, absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 73, in block_level_layout_switch
    discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 114, in block_box_layout
    absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 635, in block_container_layout
    draw_bottom_decoration)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 304, in _linebox_layout
    for i, (line, resume_at) in enumerate(lines_iterator):
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\inline.py", line 45, in iter_line_boxes
    containing_block, absolute_boxes, fixed_boxes, first_letter_style)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\inline.py", line 63, in get_next_linebox
    skip_stack = skip_first_whitespace(linebox, skip_stack)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\inline.py", line 213, in skip_first_whitespace
    result = skip_first_whitespace(box.children[index], next_skip_stack)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\inline.py", line 213, in skip_first_whitespace
    result = skip_first_whitespace(box.children[index], next_skip_stack)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\inline.py", line 199, in skip_first_whitespace
    assert next_skip_stack is None
AssertionError
```

```
(env) E:\usr3\GitHub\openpubarchive\app>python -m weasyprint --verbose ANRP.001.0007A.works.html ANRP.001.0007A.works.pdf
INFO: Step 1 - Fetching and parsing HTML - ANRP.001.0007A.works.html
INFO: Step 2 - Fetching and parsing CSS - https://cdn.jsdelivr.net/npm/fork-awesome@1.2.0/css/fork-awesome.min.css
WARNING: Ignored `font-display:block` at 12:478, descriptor not supported.
WARNING: Ignored `text-rendering:auto` at 12:585, unknown property.
WARNING: Ignored `filter:none` at 12:2847, unknown property.
INFO: Step 3 - Applying CSS
INFO: Step 4 - Creating formatting structure
INFO: Step 5 - Creating layout - Page 1
INFO: Step 6 - Creating PDF
INFO: Step 7 - Adding PDF metadata
```

I will attach the two files.  If anyone can help, I'd appreciate it!

[WeasyPrint Fail Demo.zip](https://github.com/Kozea/WeasyPrint/files/8643439/WeasyPrint.Fail.Demo.zip)

**Comment by liZe** [comment]
Hello!

It works for me:
[ANRP.001.0007A.pdf](https://github.com/Kozea/WeasyPrint/files/8645819/ANRP.001.0007A.pdf)

Could you please tell me which font is used when you render this document on your computer? It could also be useful to get the PDF you generated with the HTML file that works (and it includes the font).

**Comment by nrshapiro** [comment]
Thank you for your response.

Strange it works for you.  Here's how I'm running it for testing (I left that out above).

python -m weasyprint --verbose ANRP.001.0007A1.html ANRP.001.0007A1.pdf

using Python 3.7.4

(env) E:\usr3\GitHub\openpubarchive\app>pipdeptree -p weasyprint
weasyprint==54.3
  - cffi [required: >=0.6, installed: 1.14.6]
    - pycparser [required: Any, installed: 2.21]
  - cssselect2 [required: >=0.1, installed: 0.4.1]
    - tinycss2 [required: Any, installed: 1.1.0]
      - webencodings [required: >=0.4, installed: 0.5.1]
    - webencodings [required: Any, installed: 0.5.1]
  - fonttools [required: >=4.0.0, installed: 4.32.0]
  - html5lib [required: >=1.1, installed: 1.1]
    - six [required: >=1.9, installed: 1.15.0]
    - webencodings [required: Any, installed: 0.5.1]
  - Pillow [required: >=4.0.0, installed: 9.0.1]
  - pydyf [required: >=0.0.3, installed: 0.1.2]
  - Pyphen [required: >=0.9.1, installed: 0.10.0]
  - tinycss2 [required: >=1.0.0, installed: 1.1.0]
    - webencodings [required: >=0.4, installed: 0.5.1]

For the file that works, here are the fonts used according to Acrobat:

<img width="207" alt="image" src="https://user-images.githubusercontent.com/4837554/167295387-0378c6ce-bd01-4bbb-ae62-6c4d889e7a79.png">

[ANRP.001.0007A.works.pdf.zip](https://github.com/Kozea/WeasyPrint/files/8646931/ANRP.001.0007A.works.pdf.zip)

I'm attaching the PDF that works (zipped)

**Comment by nrshapiro** [comment]
Test2.  Same file, changes as below:

1) I deleted two blank lines.  Still fails.  See ANRP.001.0007A_Test2_Fails.html
2) I put a return after `<p class="bibentry" id="B0060">` so that the `<span>` is on the next line.  See ANRP.001.0007A_Test2_Works.html

I was preparing to get rid of the reference but just checking that this was still going to fail.  But it did not.  Both files attached, along with the PDF from ANRP.001.0007A_Test2_Works.html
[Test2files.zip](https://github.com/Kozea/WeasyPrint/files/8646961/Test2files.zip)

Run info:
```
(env) E:\usr3\GitHub\openpubarchive\app>python -m weasyprint --verbose ANRP.001.0007A_Test2_Works.html ANRP.001.0007A_Test2_Works.pdf
INFO: Step 1 - Fetching and parsing HTML - ANRP.001.0007A_Test2_Works.html
INFO: Step 2 - Fetching and parsing CSS - https://cdn.jsdelivr.net/npm/fork-awesome@1.2.0/css/fork-awesome.min.css
WARNING: Ignored `font-display:block` at 12:478, descriptor not supported.
WARNING: Ignored `text-rendering:auto` at 12:585, unknown property.
WARNING: Ignored `filter:none` at 12:2847, unknown property.
INFO: Step 3 - Applying CSS
INFO: Step 4 - Creating formatting structure
INFO: Step 5 - Creating layout - Page 1
INFO: Step 6 - Creating PDF
INFO: Step 7 - Adding PDF metadata

(env) E:\usr3\GitHub\openpubarchive\app>python -m weasyprint --verbose ANRP.001.0007A_Test2_Fails.html ANRP.001.0007A_Test2_Fails.pdf
INFO: Step 1 - Fetching and parsing HTML - ANRP.001.0007A_Test2_Fails.html
INFO: Step 2 - Fetching and parsing CSS - https://cdn.jsdelivr.net/npm/fork-awesome@1.2.0/css/fork-awesome.min.css
WARNING: Ignored `font-display:block` at 12:478, descriptor not supported.
WARNING: Ignored `text-rendering:auto` at 12:585, unknown property.
WARNING: Ignored `filter:none` at 12:2847, unknown property.
INFO: Step 3 - Applying CSS
INFO: Step 4 - Creating formatting structure
INFO: Step 5 - Creating layout - Page 1
Traceback (most recent call last):
  File "C:\Users\nrsha\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\nrsha\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\__main__.py", line 192, in <module>
    main()
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\__main__.py", line 188, in main
    html.write_pdf(output, **kwargs)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\__init__.py", line 186, in write_pdf
    counter_style=counter_style, image_cache=image_cache)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\__init__.py", line 136, in render
    optimize_size, font_config, counter_style, image_cache)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\document.py", line 930, in _render
    [Page(page_box) for page_box in page_boxes],
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\document.py", line 930, in <listcomp>
    [Page(page_box) for page_box in page_boxes],
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\__init__.py", line 133, in layout_document
    pages = list(make_all_pages(context, root_box, html, pages))
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\page.py", line 855, in make_all_pages
    page, resume_at = remake_page(i, context, root_box, html)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\page.py", line 794, in remake_page
    page_state)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\page.py", line 593, in make_page
    discard=False)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 58, in block_level_layout
    page_is_empty, absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 73, in block_level_layout_switch
    discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 114, in block_box_layout
    absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 646, in block_container_layout
    next_page)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 452, in _in_flow_layout
    fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 58, in block_level_layout
    page_is_empty, absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 73, in block_level_layout_switch
    discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 114, in block_box_layout
    absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 646, in block_container_layout
    next_page)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 452, in _in_flow_layout
    fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 58, in block_level_layout
    page_is_empty, absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 73, in block_level_layout_switch
    discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 114, in block_box_layout
    absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 646, in block_container_layout
    next_page)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 452, in _in_flow_layout
    fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 58, in block_level_layout
    page_is_empty, absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 73, in block_level_layout_switch
    discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 114, in block_box_layout
    absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 646, in block_container_layout
    next_page)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 452, in _in_flow_layout
    fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 58, in block_level_layout
    page_is_empty, absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 73, in block_level_layout_switch
    discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 114, in block_box_layout
    absolute_boxes, fixed_boxes, adjoining_margins, discard)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 635, in block_container_layout
    draw_bottom_decoration)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\block.py", line 304, in _linebox_layout
    for i, (line, resume_at) in enumerate(lines_iterator):
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\inline.py", line 45, in iter_line_boxes
    containing_block, absolute_boxes, fixed_boxes, first_letter_style)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\inline.py", line 63, in get_next_linebox
    skip_stack = skip_first_whitespace(linebox, skip_stack)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\inline.py", line 213, in skip_first_whitespace
    result = skip_first_whitespace(box.children[index], next_skip_stack)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\inline.py", line 213, in skip_first_whitespace
    result = skip_first_whitespace(box.children[index], next_skip_stack)
  File "E:\usr3\GitHub\openpubarchive\app\env\lib\site-packages\weasyprint\layout\inline.py", line 199, in skip_first_whitespace
    assert next_skip_stack is None
AssertionError
```

**Comment by liZe** [comment]
> For the file that works, here are the fonts used according to Acrobat:

Thanks, I can reproduce using DejaVu Serif.

**Comment by liZe** [comment]
I’ve finally found what’s wrong. Here’s a short example that shows the problem (the traceback is different, but the origin of the problem is the same):

```html
<div style="width: 10em"><span> <span>a</span> b</span><span>cccccccccccccccccccccccccccc</span></div>
```

Here, the line should be split between before "b", because "b" and "cccc…" are not separated, and "cccc…" is too large to fit in 10em.

What we do is:
- We render the first span. It includes a space, the second span with "a", and " b". This first span doesn’t reach the end of the line.
- When this first span is rendered, it only includes the second span and " b", because the leading space is removed (according to whitespace management in CSS).
- We then render the third span including "ccccc…". As it goes beyond the end of the line, we don’t keep it.
- We then have to remove the previous text that is bounded to this text, ie "b" in our case. In order to do that, we re-render the (already rendered) first span with a little bit less space, so that it breaks as far as it can.
- The first span is split before the "b", and that’s what we want. The second span is the first child of the first span, the text box including " b" (that we split after the space) is the second child of the first span, we then store that we split the first child before the second character of its second child.
- But that’s false. Originally, the " b" text was the third child of the first span, as there was a text box with the leading space. Unfortunately, this first text box has been removed during the first rendering, breaking the reference to the point where we finally break this span.

In the code, it means that the [`previous_resume_at` we get](https://github.com/Kozea/WeasyPrint/blob/635f1818b5db7aeaaef196fdd286eb6f1281c882/weasyprint/layout/inline.py#L797-L800) is `{0: {1: {1: None}}}` instead of `{0: {2: {1: None}}}`.

The root of the problem is that we render a box twice [in `_break_waiting_children`](https://github.com/Kozea/WeasyPrint/blob/635f1818b5db7aeaaef196fdd286eb6f1281c882/weasyprint/layout/inline.py#L595). That’s often a nice shortcut, but that’s BAD. I tried to render the original box instead of the previously rendered one, it solves this problem but it breaks 4 tests.

We already have 13 non-regression tests for line breaks, looks like we’ll add another one when this bug is fixed…

If someone is crazy enough to spend some time on this here is [the patch that doesn’t work (yet)](https://github.com/Kozea/WeasyPrint/files/8652082/break-patch.txt). I may have just made a stupid typo. But I’m afraid the problem is way harder to solve…

**Comment by nrshapiro** [comment]
Thank you very much @liZe for diagnosing and working on a patch!  I hope someone takes up the challenge. to fix it, it's a real problem for our document server.

**Comment by liZe** [comment]
@nrshapiro The bug is fixed and the commit (96f2650) actually removes a dirty hack. A +36-47 patch that includes an extra test is the kind of fixes I love 💜.

However, this fix will probably not be included in the next major version (55) that should be out this week. It’s almost too beautiful to be true and will require more real-life tests to be securely pushed on production everywhere. We’d be glad to get feedback from you (and from everyone else!) so that we’re sure that it works for real.
